src/main/scala/com/github/sbt/avro/SbtAvro.scala

```diff
@@ -340,7 +337,17 @@ object SbtAvro extends AutoPlugin {
       }
     }
 
-    cachedCompile((srcDirs ** AvroFilter).get.toSet).toSeq
+    val recordOutputFiles = if(records.isEmpty) {
+      Set.empty[File]
+    } else {
+      recompile(
+        records, outDir, out.log, strType, fieldVis, enbDecimal, useNs,
+        optionalGetters, createSetters, builder
+      )
+      (outDir ** JavaFileFilter).get.toSet
```

Comment:
Ideally, we do not want to recompile those every time since this will impact all files including the model.
I think we can integrate that in the caching mechanism so it is compiled only once

Reply:
Agreed, my patience and sbt knowledge was insufficient when writing this.